src/main/java/de/refactoringbot/refactoring/supportedrefactorings/RemoveMethodParameter.java

```diff
@@ -30,299 +34,298 @@
+					"Parameter '" + parameterToBeRemoved + "' is used inside the method '" + qualifiedMethodSignature
+							+ "' which is related to the given method (i.e., inside sub or super class)!");
 		}
-		return null;
 	}
 
 	/**
-	 * This method checks if a parameter with a specific name is used inside a
-	 * method.
-	 * 
 	 * @param method
 	 * @param paramName
-	 * @return used/notUsed
+	 * @return true if given parameter name is present in given method, false
+	 *         otherwise
 	 */
-	private boolean checkIfParameterUsed(MethodDeclaration method, String paramName) {
-		// Check if Method has body
-		if (method.getBody().isPresent()) {
-			// Get body
-			BlockStmt methodBody = method.getBody().get();
-			// Get all expressions from body
-			List<NameExpr> expressions = methodBody.findAll(NameExpr.class);
-			// Iterate expressions
+	private boolean isParameterUsed(MethodDeclaration method, String paramName) {
+		Optional<BlockStmt> methodBody = method.getBody();
+		if (methodBody.isPresent()) {
+			List<NameExpr> expressions = methodBody.get().findAll(NameExpr.class);
 			for (NameExpr expression : expressions) {
-				// If expression name = param name
 				if (expression.getNameAsString().equals(paramName)) {
-					// Parameter used
 					return true;
 				}
 			}
 		}
 
-		// Parameter not used
 		return false;
 	}
 
 	/**
-	 * This method gets the Method signature that our methods will have without the
-	 * parameter.
+	 * Check if post refactoring signature already exists in given class or
+	 * interface
 	 * 
-	 * @param refactoring
-	 * @param paramName
-	 * @return signature
-	 * @throws FileNotFoundException
+	 * @param currentClassOrInterface
+	 * @param postRefactoringSignature
 	 * @throws BotRefactoringException
 	 */
-	private String getPostRefactoringSignature(ParserRefactoring refactoring, String paramName)
-			throws FileNotFoundException, BotRefactoringException {
-		for (String javaFile : refactoring.getJavaFiles()) {
-			// Create compilation unit
-			FileInputStream methodPath = new FileInputStream(javaFile);
-			CompilationUnit compilationUnit = LexicalPreservingPrinter.setup(JavaParser.parse(methodPath));
-
-			// Get all Methods and MethodCalls of File
-			List<MethodDeclaration> fileMethods = compilationUnit.findAll(MethodDeclaration.class);
-
-			// Rename all Methods
-			for (MethodDeclaration fileMethod : fileMethods) {
-				if (refactoring.getMethods().contains(fileMethod)) {
-					performRemoveMethodParameter(fileMethod, paramName);
-					return RefactoringHelper.getMethodSignatureAsString(fileMethod);
+	private void validateClassOrInterfaceNotContainsPostRefactoringSignatureAlready(
+			ClassOrInterfaceDeclaration currentClassOrInterface, String postRefactoringSignature)
+			throws BotRefactoringException {
+		if (RefactoringHelper.isLocalMethodSignaturePresentInClassOrInterface(currentClassOrInterface,
+				postRefactoringSignature)) {
+			throw new BotRefactoringException(
```

Comment:
This error message reads strangely,

I think something like "Renaming of method would lead to a duplicate method signature inside 'this class or interface'." would be easier to understand.

Or maybe add some words like "....result in [a] method signature [that is] already present [inside] the class/interface..."